scipy/spatial/distance.py

```diff
@@ -864,7 +864,7 @@ def jaccard(u, v, w=None):
     u = _validate_vector(u)
     v = _validate_vector(v)
 
-    nonzero = np.bitwise_or(u != 0, v != 0)
+    nonzero = np.bitwise_or(u.astype(bool), v.astype(bool))
```

Comment:
This unconditionally copies `u` and `v`, better to use `copy=False`:
```suggestion
    nonzero = np.bitwise_or(u.astype(bool, copy=False),
                            v.astype(bool, copy=False))
```